Services.Test/Storage/StorageRecordsTest.cs

```diff
@@ -0,0 +1,333 @@
+                    It.IsAny<StorageConfig>())
+            ).ReturnsAsync(this.mockDocumentClient.Object);
+
+            this.mockDocumentDbWrapper.Setup(
+                x => x.ReadAsync(
+                    It.IsAny<IDocumentClient>(), 
+                    It.IsAny<StorageConfig>(), 
+                    It.IsAny<string>())
+            ).ReturnsAsync(this.mockStorageDocument.Object);
+
+            this.target = new StorageRecords(
+                this.mockDocumentDbWrapper.Object, 
+                this.mockLogger.Object, 
+                this.mockInstance.Object,
+                this.mockConcurrencyConfig.Object);
+
+            this.storageConfig = new StorageConfig();
+            this.target.Init(this.storageConfig);
+        }
+
+        [Fact]
+        public void ItReturnsASingleStorageRecordThatHasNotExpired()
+        {
+            // Arrange
+            var id = "123";
+
+            // Mock a storage document that will be returned
+            var document = new Document();
+            document.Id = id;
+            document.SetPropertyValue("ExpirationUtcMsecs", Now + testOffsetMs);
+            this.mockStorageDocument.SetupGet(x => x.Resource).Returns(document);
+
+            // Act
+            Task<StorageRecord> storageRecordTask = this.target.GetAsync(id);
+            storageRecordTask.Wait(testTimeout);
+            
+            // Assert
+            Assert.Equal(id, storageRecordTask.Result.Id);
+        }
+
+        [Fact]
+        public void ItThrowsResourceNotFoundForExpiredRecords()
+        {
+            // Arrange
+            var id = "123";
+
+            // Mock a storage document that will be returned
+            var document = new Document();
+            document.Id = id;
+            document.SetPropertyValue("ExpirationUtcMsecs", 0);
+            this.mockStorageDocument.SetupGet(x => x.Resource).Returns(document);
+
+            // Act, Assert
+            var ex = Assert.Throws<AggregateException>(() =>
+            {
+                Task resultTask = this.target.GetAsync(id);
+                resultTask.Wait(testTimeout);
+            });
+            Assert.IsType<ResourceNotFoundException>(ex.InnerException);
+        }
+
+        [Fact]
+        public void ItReturnsTrueForRecordsThatExist()
+        {
+            // Arrange
+            var id = "123";
+
+            // Mock a storage document that will be returned
+            var document = new Document();
+            document.Id = id;
+            document.SetPropertyValue("ExpirationUtcMsecs", Now + testOffsetMs);
+            this.mockStorageDocument.SetupGet(x => x.Resource).Returns(document);
+
+            // Act
+            Task<bool> existsAsyncTask = this.target.ExistsAsync(id);
+            existsAsyncTask.Wait(testTimeout);
+
+            // Assert
+            Assert.True(existsAsyncTask.Result);
+        }
+
+        [Fact]
+        public void ItReturnsFalseForExpiredRecords()
```

Comment:
the test name should mention that it's checking for "exist"

Reply:
Fixed.